docs/data/joy/guides/overriding-component-structure/overriding-component-structure.md

```diff
@@ -0,0 +1,114 @@
+# Overriding component structure
+
+<p class="description">Learn how to override the default DOM structure of Joy UI components.</p>
+
+Joy UI components are designed to suit the widest possible range of use cases, but you may occasionally need to change how a component's structure is rendered in the DOM.
+
+To understand how to do this, it helps to have an accurate mental model of MUI components.
+
+## The mental model
+
+A component's structure is determined by the elements that fill that component's **slots**.
+Slots are most commonly filled by HTML tags, but may also be filled by React components.
+
+All components contain a root slot that defines their primary node in the DOM tree; more complex components also contain additional interior slots named after the elements they represent.
+
+All _non-utility_ Joy UI components accept two props for overriding their rendered HTML structure:
+
+- `component`—to override the root slot
+- `slots`—to override any interior slots (when present) as well as the root
+
+Additionally, you can pass custom props to interior slots using `slotProps`.
+
+## The root slot
+
+The root slot represents the component's "primary" element.
+For simpler components, the root slot is often filled by the native HTML element that the component is intended to replace.
+
+For example, the [Button's](/joy-ui/react-button/) root slot is a `<button>` element.
+This component _only_ has a root slot; more complex components may have additional [interior slots](#interior-slots).
+
+### The component prop
+
+Use the `component` prop to override a component's root slot.
+The demo below shows how to replace the Button's `<button>` tag with a `<div>`:
+
+{{"demo": "OverridingRootSlot.js"}}
+
+:::info
+If you provide a non-interactive element like a `<div>` or a `<span>`, the Button will automatically add the necessary accessibility attributes.
+Try inspecting the demo Button above in your browser's dev tools to see this feature in action.
+:::
+
+## Interior slots
+
+Complex components are composed of one or more interior slots in addition to the root.
+These slots are often (but not necessarily) nested within the root.
+
+For example, the [Slider](/joy-ui/react-slider/) is composed of a root `<span>` that houses several interior slots named for the elements they represent: track, thumb, rail, and so on.
+
+### The slots prop
+
+Use the `slots` prop to override a component's interior slots.
+The example below shows how to override the listbox slot in the [Select](/joy-ui/react-select/) component—a `<ul>` by default—with an `<ol>`:
+
+{{"demo": "OverridingInternalSlot.js"}}
```

Comment:
I think using a simplified version of https://mui.com/joy-ui/react-autocomplete/#githubs-picker could make it easier to understand because it is a real use case.